src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -204,6 +204,17 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="GetLatestCommitHash" Condition="'$(LatestCommit)'==''">
+    <Exec Command="git rev-parse HEAD" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="LatestCommit" />
+      <Output TaskParameter="ExitCode" PropertyName="LatestCommitExitCode" />
+    </Exec>
+    <!-- We shouldn't fail the build if we can't retreive the commit hash, so in this case just set it to N/A -->
+    <PropertyGroup Condition="'$(LatestCommitExitCode)'!='0'">
+      <LatestCommit>N/A</LatestCommit>
```

Comment:
Yes, if you see the target's condition, it will only execute if you haven't defined `LatestCommit` already, for example in official builds this will be already passed in. This error handling is more for the case were LatestCommit is not defined, and the git rev-parse command fails, which may happen if git is not installed or not in the path for example.